OpenRA.Game/Game.cs

```diff
@@ -190,26 +189,18 @@ public static void RestartGame()
 			if (replay != null)
 				JoinReplay(replayName);
 			else
-				StartMission(uid, globalSettings.GameSpeedType, globalSettings.Difficulty);
-		}
-
-		public static void StartMission(string mapUID, string gameSpeed, string difficulty, Action onStart = null)
-		{
-			OrderManager om = null;
-
-			Action lobbyReady = null;
-			lobbyReady = () =>
 			{
-				LobbyInfoChanged -= lobbyReady;
-				om.IssueOrder(Order.Command("gamespeed {0}".F(gameSpeed)));
-				om.IssueOrder(Order.Command("difficulty {0}".F(difficulty)));
-				om.IssueOrder(Order.Command("state {0}".F(Session.ClientState.Ready)));
-				if (onStart != null)
-					onStart();
-			};
-			LobbyInfoChanged += lobbyReady;
+				LobbyInfoChanged += WidgetUtils.Once(() =>
```

Comment:
Please use the same pattern as `StartMission` here: Bind to LobbyInfo and then remove the binding from inside the callback before it returns.